src/main/webapp/resources/js/components/project/ProjectSelect.tsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import { Select, SelectProps, Tag, Typography } from "antd";
+
+export type Project = { identifier: number; name: string };
+
+export interface ProjectSelectProps extends SelectProps {
+  projects: Project[];
+}
+
+/**
+ * React component for displaying a project drop-down menu.
+ * @param {list} projects - list of projects that is to be displayed
+ * @param {function} onChange - function that is called when select option has changed
+ * @param {number} defaultValue - project identifier of the project that is to be displayed by default
+ * @returns {JSX.Element}
+ * @constructor
+ */
+export function ProjectSelect({
+  projects,
+  onChange = undefined,
+  defaultValue = null,
+}: ProjectSelectProps): JSX.Element {
+  const [options, setOptions] = React.useState(() => formatOptions(projects));
```

Comment:
You need to state what type the options are. I think the best way that I have come across is:

```typescript
const [options, setOptions] = React.useState<SelectOptions[]>(()=>formatOptions(projects));
```

Reply:
Please see 7bfbc4eadec7175b69d3be65bc1cc282ba116ae5.